icechunk/src/repository.rs

```diff
@@ -597,15 +609,8 @@ impl Repository {
     }
 
     pub async fn clear(&mut self) -> RepositoryResult<()> {
-        let to_delete: Vec<(NodeType, Path)> =
-            self.list_nodes().await?.map(|node| (node.node_type(), node.path)).collect();
-
-        for (t, p) in to_delete {
-            match t {
-                NodeType::Group => self.delete_group(p).await?,
-                NodeType::Array => self.delete_array(p).await?,
-            }
-        }
+        #[allow(clippy::expect_used)]
+        self.delete_group(Path::try_from("/").expect("should not fail")).await?;
```

Comment:
Is this too clever? I think the root group doesn't need to exist, for example they can have a root array.